funcs/func_curl.c

```diff
@@ -698,7 +729,7 @@ static int acf_curl_helper(struct ast_channel *chan, struct curl_args *args)
 		if (cur->key == CURLOPT_SPECIAL_HASHCOMPAT) {
 			hashcompat = (long) cur->value;
 		} else if (cur->key == CURLOPT_HTTPHEADER) {
-			headers = curl_slist_append(headers, (char*) cur->value);
+			replace_or_add_header(&headers, (char*) cur->value);
```

Comment:
I do not believe it should be the responsibility of the acf_curl_helper function to do this replacement. The list of headers on the datastore should only ever contain a single header of the given name. The global headers list should not add a header to the request if the same header is present in the datastore list.

Reply:
so you want the change in acf_curlopt_write ath the end where 	AST_LIST_INSERT_TAIL(list, new, list); ?

Reply:
That is where I would expect such functionality to be written, but there would also have to be logic in acf_curl_helper for the global / channel datastore situation.

Reply:
ok sorry cant understand what you want, you can close this if you feel its not good enough as I don't have the knowledge to do it as you want.